src/actions/requests.rs

```diff
@@ -456,22 +462,12 @@ impl<'a> RequestAction<'a> for Rename {
 
 pub struct Deglob;
 
-impl<'a> Action<'a> for Deglob {
-    type Params = Location;
-    const METHOD: &'static str = "rustWorkspace/deglob";
-
-    fn new(_: &'a mut LsState) -> Self {
-        Deglob
-    }
-}
-
-impl<'a> RequestAction<'a> for Deglob {
-    type Response = Ack;
-    fn handle<O: Output>(&mut self, id: usize, location: Self::Params, ctx: &mut ActionContext, out: O) -> Result<Self::Response, ()> {
+impl Deglob {
+    fn calculate_workspace_edit(&mut self, location: &<Self as Action>::Params, ctx: &ActionContext) -> Result<DeglobResult, (ErrorCode, &'static str)> {
```

Comment:
I believe the `&mut self` is unnecessary here

Reply:
Yes, copy and paste error from handle